homeassistant/components/icloud/__init__.py

```diff
@@ -1 +1,593 @@
+                interval = self._max_interval
+
+            if (
+                device.battery_level is not None
+                and device.battery_level <= 33
+                and mindistance > 3
+            ):
+                # Low battery - let's check half as often
+                interval = interval * 2
+
+            intervals[device.name] = interval
+
+        return max(
+            int(min(intervals.items(), key=operator.itemgetter(1))[1]),
+            self._max_interval,
+        )
+
+    def keep_alive(self, now):
+        """Keep the API alive."""
+        if self.api is None:
+            self.reset_account()
+
+        if self.api is None:
+            return
+
+        self.api.authenticate()
+        self.update_devices()
+
+    @property
+    def name(self):
+        """Return the account name."""
+        return self._accountname
+
+
+class IcloudDevice(Entity):
+    """Representation of a iCloud device."""
+
+    def __init__(self, account: IcloudAccount, device: AppleDevice):
+        """Initialize the iCloud device."""
+        self.hass = account.hass
+        self._account = account
+        self._accountname = account.name
+        self._accountname_slug = slugify(self._accountname.partition("@")[0])
+
+        self._device = device
+        self._status = device.status(DEVICE_STATUS_SET)
+        _LOGGER.debug("Device Status is %s", self._status)
+
+        self._name = self._status["name"]
+        self._dev_id = slugify(self._name.replace(" ", "", 99))  # devicename
+        self._unique_id = f"{self._accountname_slug}_{self._dev_id}"
+        self._device_class = self._status["deviceClass"]
+        self._device_name = self._status["deviceDisplayName"]
+        if self._status["prsId"]:
+            self._owner_fullname = account.family_members_fullname[
+                self._status["prsId"]
+            ]
+        else:
+            self._owner_fullname = account.account_owner_fullname
+
+        self._battery_level = None
+        self._battery_status = None
+        self._low_power_mode = None
+        self._location = None
+
+        self._seen = False
+
+        self.update(self._status)
+
+    def update(self, status):
+        """Update the iCloud device."""
+        self._status = status
+
+        self._device_status = DEVICE_STATUS_CODES.get(
+            self._status["deviceStatus"], "error"
+        )
+
+        self._attrs = {
+            CONF_ACCOUNTNAME: self._accountname,
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_DEVICENAME: self._device_name,
+            ATTR_DEVICESTATUS: self._device_status,
+            ATTR_OWNERNAME: self._owner_fullname,
+        }
+
+        if self._status["batteryStatus"] != "Unknown":
+            self._battery_level = round(self._status.get("batteryLevel", 0) * 100)
+            self._battery_status = self._status["batteryStatus"]
+            self._low_power_mode = self._status["lowPowerMode"]
```

Comment:
This is only used within this method. It could be a local variable instead of an instance attribute.